### PR TITLE
Add more stream properties to new_stream

### PR DESCRIPTION
**Pull request by thiagocaiubi**
I wrote the ones that I'm missing right now but I can write all of them if you agree about this kind of approach. 

Reference: [http://dev.librato.com/v1/instruments](http://dev.librato.com/v1/instruments) (Stream Properties section)

TODO
 - [x] Validate WIP with maintainers
 - [x] Add support for group_function
 - [x] Add support for color
 - [x] Add support for name
 - [x] Enhance composite validation based on:
 - `A composite metric query string to execute when this stream is displayed. This can not be specified with a metric, source or group_function.`
 - [x] Update README.md

### CODE REVIEW

**Comment by drio** [comment]
💥 I like where this is going. Thanks @thiagocaiubi.

A minor comment: I'd explore alternatives to avoid having such a long 
list of parameters.

Also, can you expand on this?

```
A composite metric query string to execute when this stream is displayed. 
This can not be specified with a metric, source or group_function.
```

Good work!

**Comment by thiagocaiubi** [comment]
@drio This is it! :smile: 

I've tried to reduce that long parameters list everywhere but some still exist, please provide feedback on this.

In order to validate the coexisting properties `composite`, `metric`, `source` and `group_function` I decided to keep backward compatibility and just omit conflicting properties. I'm not sure about this approach,  maybe it is better to give a feedback to users (just like the API does) instead of make decisions in their place. But I also thought that could be an another PR.

File: librato/instruments.py [hunk 1]
```diff
@@ -55,14 +55,53 @@ def save(self):
 
 
 class Stream(object):
-    def __init__(self, metric=None, source='*', composite=None):
+    def __init__(self, metric=None, source='*', composite=None, name=None,
+            units_short=None, units_long=None, display_min=None, display_max=None,
+            summary_function='average', transform_function=None, period=None,
+            group_function='average', color=None):
```
Comment (drio) [inline librato/instruments.py]: You can explore the option of iterating over the named arguments and set them in the object based on a class property that contains the valid list of stream attributes.

File: librato/instruments.py [hunk 2]
```diff
@@ -55,14 +55,53 @@ def save(self):
...
+        self.transform_function = transform_function
+        self.period = period
+        self.group_function = group_function
+        self.color = color
+
+        # Omit conflicting properties (http://dev.librato.com/v1/instruments)
         if self.composite:
+            self.metric = None
             self.source = None
+            self.group_function = None
```
Comment (drio) [inline librato/instruments.py]: I am fine with doing this but we should let the user know so she is aware.

**Comment by nextmat** [comment]
@drio Just a heads up, chart streams have slightly different properties than instrument streams, in case it is relevant for this PR.

**Comment by jderrett** [comment]
Closing this in favor of #116, thanks a ton @thiagocaiubi!

**Pull request closed without merging**